src/config-import.js

```diff
@@ -0,0 +1,277 @@
+              name,
+              variants: [variant],
+            }
+          } else if (icon.name === name) {
+            icon.variants.push(variant)
+          } else {
+            yield icon
+            icon = {
+              name,
+              variants: [variant],
+            }
+          }
+        } catch (err) {
+          warnings.push(err)
+        }
+      }
+      if (icon) {
+        yield icon
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ name: string, value: import('@mapeo/schema').FieldValue }>}
+     */
+    *fields() {
+      const { fields } = presetsFile
+      for (const [name, field] of Object.entries(fields)) {
+        if (!isRecord(field) || !hasOwn(field, 'key')) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        /** @type {Record<string, unknown>} */
+        const fieldValue = {
+          schemaName: 'field',
+          tagKey: field.key,
+        }
+        for (const key of Object.keys(valueSchemas.field.properties)) {
+          if (hasOwn(field, key)) {
+            // @ts-ignore - we validate below
+            fieldValue[key] = field[key]
+          }
+        }
+        if (!validate('field', fieldValue)) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        yield {
+          name,
+          value: fieldValue,
+        }
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ fieldNames: string[], iconName: string | undefined, value: import('@mapeo/schema').PresetValue }>}
+     */
+    *presets() {
+      const { presets } = presetsFile
+      // sort presets using the sort field, turn them into an array
+      const sortedPresets = Object.keys(presets)
+        .filter((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          const isInvalidPreset = !isRecord(preset)
+          if (isInvalidPreset) {
+            warnings.push(new Error(`invalid preset ${presetName}`))
+          }
+          return !isInvalidPreset
+        })
+        .map((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          return preset
+        })
+        .sort((preset, nextPreset) => {
+          let sort = preset.sort || Infinity
+          let nextSort = nextPreset.sort || Infinity
+          return sort - nextSort
+        })
+        .map((preset) => {
+          delete preset.sort
+          return preset
+        })
+
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let preset of sortedPresets) {
+        /** @type {any} */
+        const presetValue = {
+          schemaName: 'preset',
+          fieldIds: [],
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
```

Comment:
We can get some type safety (by removing `any`s and `ts-ignore`s) with some minor changes:

```suggestion
      /** @type {Array<Record<string, unknown>>} */
      const sortedPresets = []
      for (const [presetName, preset] of Object.entries(presets)) {
        if (isRecord(preset)) {
          sortedPresets.push(preset)
        } else {
          warnings.push(new Error(`invalid preset ${presetName}`))
        }
      }
      sortedPresets.sort((preset, nextPreset) => {
        const sort = typeof preset.sort === 'number' ? preset.sort : Infinity
        const nextSort =
          typeof nextPreset.sort === 'number' ? nextPreset.sort : Infinity
        return sort - nextSort
      })

      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
      for (let preset of sortedPresets) {
        /** @type {Record<string, unknown>} */
        const presetValue = {
          schemaName: 'preset',
          fieldIds: [],
          addTags: {},
          removeTags: {},
          terms: [],
        }
        for (const key of Object.keys(valueSchemas.preset.properties)) {
          if (hasOwn(preset, key)) {
```

(This suggestion looks larger than it really because it doesn't show the proper diff...)